src/core_modules/capture-core/components/Pages/MainPage/WorkingLists/ListViewUpdater/ListViewUpdater.component.js

```diff
@@ -1,56 +1,68 @@
 // @flow
-import React from 'react';
+import React, { useEffect, useRef, useContext } from 'react';
 import log from 'loglevel';
 import { errorCreator } from 'capture-core-utils';
 import { ListViewUpdaterContext } from '../workingLists.context';
 import { ListViewBuilder } from '../ListViewBuilder';
 import { areFiltersEqual } from '../utils';
 import type { Props } from './listViewUpdater.types';
 
-function useUpdateListMemoize(value) {
+const useUpdateListMemoize = (value) => {
     const [filters, ...rest] = value;
-    const filtersRef = React.useRef(filters);
+    const filtersRef = useRef(filters);
     const prevFilters = filtersRef.current;
 
     if (!areFiltersEqual(prevFilters, filters)) {
         filtersRef.current = filters;
     }
 
     return [filtersRef.current, ...rest];
-}
+};
 
-function useUpdateListEffect(callback, dependencies) {
-    const firstRun = React.useRef(true);
+const useUpdateEffect = (callback, { forceFirstRunUpdate, filters, restDependencies }) => {
```

Comment:
I think you not using `filters` inside this function. Couldnt they be part of the rest dependencies?

Reply:
ah okay I dint read that well